### PR TITLE
store/copr: invalidate stale regions for Mpp query. (#24410)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #24410 to release-5.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In tidb repo:
git pr https://github.com/pingcap/tidb/pull/24432
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/tidb.git pr/24432:release-5.0-c269b117054a
```

---

### What problem does this PR solve?

Problem Summary:
Right now, we won't update the stale regions during executing Mpp queries. Then we added the try regions field in MppDispatchTask reponse. In this pr, we try to invalidate the stale regions in the mppr response.

### Related changes
https://github.com/pingcap/kvproto/pull/751

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
  - Insert data continously
  - observe that tiflash is retrying regions. In my case, the log is like `["DAGQueryBlockInterpreter: Start to retry 1 regions (10936,)"]`
  - observe log in tidb, find `["invalid region because tiflash detected stale region"] ["region id"="{ region id: 10936, ver: 116, confVer: 2 }"]` proving this function works.
 

### Release note 

- store/copr: invalidate stale regions for Mpp query.

### CODE REVIEW

File: store/copr/mpp.go [hunk 1]
```diff
@@ -243,6 +243,13 @@ func (m *mppIterator) handleDispatchReq(ctx context.Context, bo *tikv.Backoffer,
 		m.sendError(errors.New(realResp.Error.Msg))
 		return
 	}
+	if len(realResp.RetryRegions) > 0 {
```
Comment (xiongjiwei) [inline store/copr/mpp.go:246]: why we need this `if`

Comment (hanfei1991) [reply store/copr/mpp.go:246]: It's unnecessary, I will fix it

**Review by xiongjiwei — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**